src/lib/y2packager/clients/inst_product_upgrade_license.rb

```diff
@@ -64,13 +64,19 @@ def product
         @product = Y2Packager::Product.selected_base
         # restore the initial status, the package update will be turned on later again
         Yast::Pkg.PkgReset
-        changed = Yast::Pkg.ResolvableProperties("", :package, "").select do |p|
-          p["status"] != :available || p["status"] != :installed
-        end
-        changed.each { |p| Yast::Pkg.PkgNeutral(p["name"]) }
+        reset_packages(:removed)
+        reset_packages(:selected)
 
         @product
       end
+
+      # Reset packages having the specific status
+      # @param status [Symbol] Status symbol, `:removed`, `:selected`, `:installed`, `:available`
+      # @see Yast::Pkg.GetPackages
+      def reset_packages(status)
+        # packages in the required status
+        Yast::Pkg.GetPackages(status, true).each { |p| Yast::Pkg.PkgNeutral(p) }
```

Comment:
What does this `true` mean?
Do we have an online API reference for `Pkg`?

Reply:
OK, this is it https://yast-pkg-bindings.surge.sh/classYast_1_1Pkg.html#a62a42363467a84326560adef5f795ec1

Reply:
For extra clarity, I would use
```rb
names_only = true
...GetPackages(status, names_only)...
```